src/json.hpp.re2c

```diff
@@ -618,7 +700,8 @@ class basic_json
         boolean,        ///< boolean value
         number_integer, ///< number value (integer)
         number_float,   ///< number value (floating-point)
-        discarded       ///< discarded by the the parser callback function
+        discarded,      ///< discarded by the the parser callback function
+        number_unsigned ///< number value (unsigned integer)
```

Comment:
Hm. The values of the enum are actually available via public member functions. As this may break existing code, then the extension should be in the 2.0.0 release. I'd rather have clean code than something ugly just to have this edge case covered by a 1.x.x release.

About the tests: it would be great to convince Valgrind to accept the code. If this is some kind of bug, then you give these tests a label so they can be skipped when run with Valgrind...